assets/js/utils/channelUtils.ts

```diff
@@ -144,6 +144,37 @@ export const syncPresenters = (
   }
 };
 
+export const isClientAPresenter = (client: Client, presenters: Presenter[]) => {
+  return presenters.some((presenter) => presenter.email === client.email);
+};
+
+export const switchAskingForBeingPresenter = (
+  eventChannel: Channel | undefined,
+  email: string,
+  isAlreadyAsking: boolean
+): boolean => {
+  return isAlreadyAsking
+    ? stopAskingForBeingPresenter(eventChannel, email)
+    : askForBeingPresenter(eventChannel, email);
+};
+
+export const askForBeingPresenter = (eventChannel: Channel | undefined, email: string): boolean => {
+  pushToChannel(eventChannel, email, "presenting_request");
+  return true;
+};
+
+export const stopAskingForBeingPresenter = (
+  eventChannel: Channel | undefined,
+  email: string
+): boolean => {
+  pushToChannel(eventChannel, email, "cancel_presenting_request");
+  return false;
+};
```

Comment:
You could simplify this expression to:
```ts
export const switchAskingForBeingPresenter = (
  eventChannel: Channel | undefined,
  email: string,
  isAlreadyAsking: boolean
): boolean => {
  const message = isAlreadyAsking ? "cancel_presenting_request" : "presenting_request";
  
  pushToChannel(eventChannel, email, message);
  
  return !isAlreadyAsking;
};
```

BUT this function mutates the state by sending event to channel and it returns a value. I'm not a fan of such behavior. It breaks the CQS rule (command query separation). I suggest rewriting it to:

```ts
export const switchAskingForBeingPresenter = (
  eventChannel: Channel | undefined,
  email: string,
  isAlreadyAsking: boolean
) => pushToChannel(eventChannel, email, isAlreadyAsking ? "cancel_presenting_request" : "presenting_request");
```

There is no return statement. Now it does only one thing :) 

This function is used in two places. This one doesn't require change.

```ts
const handleClick = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    const shouldAskForBeingPresenter = (e.target as HTMLTextAreaElement).value === askText;

    switchAskingForBeingPresenter(eventChannel, participant.email, shouldAskForBeingPresenter);
  };
```

and this one is easy to fix. From:

```ts
switchAsking={(isAsking) => {
     return switchAskingForBeingPresenter(eventChannel, client.email, isAsking);
}}
```

to: 

```ts
switchAsking={(isAsking) => {
  switchAskingForBeingPresenter(eventChannel, client.email, isAsking);
  return !isAsking;
}}
```

Reply:
but we can go even further and remove return `!isAsking` from the last code snipped and do something like that:

```ts
 const [isAskingForPresenter, setIsAskingForPresenter] = useState<Boolean>(false);

...


GenericButton
        icon={<UserPlus className="PanelButton" />}
        onClick={() => {
          switchAsking(isAskingForPresenter);
          setIsAskingForPresenter(!isAskingForPresenter);
          getInfoToast(toast, toastText(newAskingState), 2_000);
        }}
        className={className}
      />
```

and now we don't need any returns in `switchAsking` and `switchAskingForBeingPresenter`